helix-term/src/commands.rs

```diff
@@ -1596,14 +1596,7 @@ fn replace(cx: &mut Context) {
                 if !range.is_empty() {
                     let text: String =
                         RopeGraphemes::new(doc.text().slice(range.from()..range.to()))
-                            .map(|g| {
-                                let cow: Cow<str> = g.into();
-                                if str_is_line_ending(&cow) {
-                                    cow
-                                } else {
-                                    ch.into()
-                                }
-                            })
+                            .map(|_| -> Cow<str> { ch.into() })
                             .collect();
 
                     (range.from(), range.to(), Some(text.into()))
```

Comment:
We can skip collecting into a String and build the Tendril directly using Tendril's `FromIterator` for `&str`s:

```rust
let text: Tendril = RopeGraphemes::new(doc.text().slice(range.from()..range.to())).map(|_g| ch).collect();
(range.from(), range.to(), Some(text))
```